src/bp/nlu-core/intents/oos-intent-classfier.ts

```diff
@@ -0,0 +1,245 @@
+    const nbOfNoneUtterances = lo.clamp(
+      (allUtterances.length * 2) / 3,
+      NONE_UTTERANCES_BOUNDS.MIN,
+      NONE_UTTERANCES_BOUNDS.MAX
+    )
+    const stopWords = await getStopWordsForLang(languageCode)
+    const vocabWords = lo(allTokens)
+      .filter(t => t.tfidf <= SMALL_TFIDF)
+      .map(t => t.toString({ lowerCase: true }))
+      .uniq()
+      .orderBy(t => t)
+      .value()
+
+    // If 30% in utterances is a space, language is probably space-separated so we'll join tokens using spaces
+    const joinChar = vocabWithDupes.filter(x => isSpace(x)).length >= vocabWithDupes.length * 0.3 ? SPACE : ''
+
+    const vocabUtts = lo.range(0, nbOfNoneUtterances).map(() => {
+      const nbWords = Math.round(lo.random(1, avgTokens * 2, false))
+      return lo.sampleSize(lo.uniq([...stopWords, ...vocabWords]), nbWords).join(joinChar)
+    })
+
+    const junkWordsUtts = lo.range(0, nbOfNoneUtterances).map(() => {
+      const nbWords = Math.round(lo.random(1, avgTokens * 2, false))
+      return lo.sampleSize(junkWords, nbWords).join(joinChar)
+    })
+
+    const mixedUtts = lo.range(0, nbOfNoneUtterances).map(() => {
+      const nbWords = Math.round(lo.random(1, avgTokens * 2, false))
+      return lo.sampleSize([...junkWords, ...stopWords], nbWords).join(joinChar)
+    })
+
+    return <Intent<Utterance>>{
+      name: NONE_INTENT,
+      slot_definitions: [],
+      utterances: await buildUtteranceBatch(
+        [...mixedUtts, ...vocabUtts, ...junkWordsUtts, ...stopWords],
+        languageCode,
+        this.tools
+      ),
+      contexts: []
+    }
+  }
+
+  private async _trainOOScopeSvm(
+    trainInput: TrainInput,
+    noneIntent: Omit<Intent<Utterance>, 'contexts'>,
+    progress: (p: number) => void
+  ): Promise<string | undefined> {
+    const { allUtterances, nluSeed, intents } = trainInput
+    const { languageCode } = allUtterances[0]
+
+    const trainingOptions: MLToolkit.SVM.SVMOptions = {
+      c: [10], // so there's no grid search
+      kernel: 'LINEAR',
+      classifier: 'C_SVC',
+      seed: nluSeed
+    }
+
+    const noneUtts = noneIntent.utterances
+
+    if (!isPOSAvailable(languageCode) || noneUtts.length === 0) {
+      progress(1)
+      return
+    }
+
+    const vocab = this.getVocab(allUtterances)
+    const oos_points = featurizeOOSUtterances(noneUtts, vocab, this.tools)
+
+    const in_ctx_scope_points = _.chain(intents)
+      .filter(i => i.name !== NONE_INTENT)
+      .flatMap(i => featurizeInScopeUtterances(i.utterances, i.name))
+      .value()
+
+    const svm = new this.tools.mlToolkit.SVM.Trainer()
+
+    const model = await svm.train([...in_ctx_scope_points, ...oos_points], trainingOptions, progress)
+    return model
+  }
+
+  private async _trainInScopeSvm(
+    trainInput: TrainInput,
+    noneIntent: Omit<Intent<Utterance>, 'contexts'>,
+    progress: (p: number) => void
+  ): Promise<string> {
+    const baseIntentClf = new BaseIntentClassifier(this.tools)
```

Comment:
I really don't like that you call this BaseIntentClassifier. why not just IntentClassifier the Base keyword doesn't really give context to me it implies that some classes are inheriting but in fact, not really

Reply:
Sure, I will rename in upcoming PRs. I currently have it called `SvmIntentClassifier` in local, but this name doesnt really suits either. Just `IntentClassifier` is fine.